src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1989,6 +2001,51 @@ def make_api_call_with_retries(uri, data, tls_verify, fault_type, summary, cli_e
                 close_subprocess_and_raise_cli_error(clientproxy_process, cli_error + str(e))
 
 
+def fetch_pop_publickey_kid(api_server_port, clientproxy_process):
+    requestbody = {}
+    poppublickey_uri = f'https://localhost:{api_server_port}/identity/poppublickey'
+    # Needed to prevent skip tls warning from printing to the console
+    original_stderr = sys.stderr
+    f = open(os.devnull, 'w')
+    sys.stderr = f
+
+    get_publickey_response = make_api_call_with_retries(poppublickey_uri, requestbody, "get", False, consts.Get_PublicKey_Info_Fault_Type,
+                                                        'Unable to fetch public key info from clientproxy',
+                                                        "Failed to fetch public key info from client proxy", clientproxy_process)
+
+    sys.stderr = original_stderr
+    publickey_info = json.loads(get_publickey_response.text)
```

Comment:
Do we get TP also as part of response which we can reuse?

Reply:
I meant `req_cnfJSON = {"kid": kid, "xms_ksl": "sw"}` part which we create on our side. So was curious about response.

Reply:
Got it. We don't get that as part of the response